### PR TITLE
Add cross-compile images for Alpine ARM64

### PR DESCRIPTION
**Pull request by aqrln**
Add cross-compile images for ARM64 musl.

Ref: https://github.com/prisma/prisma/issues/8478
Ref: https://github.com/prisma/engineer/pull/87

### CODE REVIEW

File: cross/aarch64-musl/musl-clang [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+LIBPREFIX=/usr/lib/aarch64-linux-musl
+GCCLIBPREFIX=/usr/lib/gcc-cross/aarch64-linux-gnu/10
+
+if echo " $@" | fgrep -q -- ' -pie'; then
+  CRT=${LIBPREFIX}/Scrt1.o
+else
+  CRT=${LIBPREFIX}/crt1.o
+fi
+
+exec clang-13 -target aarch64-linux-musl \
```
[the remark this answers is not in the document]
Comment (aqrln) [reply cross/aarch64-musl/musl-clang:14]: also, for posterity, one of the options I considered with was using zig as the c toolchain replacement, but it didn't really work as well as I hoped and there were some issue during linking so I gave up on that idea

File: cross/linux-musl-arm.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+ENV OPENSSL_DIR=/opt/cross
+
+# Unset CC and CXX variables used during cross-compilation of zlib and
+# openssl. `CC_aarch64_unknown_linux_musl` and `CXX_aarch64_unknown_linux_musl`,
+# which are necessary for cross-compilation of C code in sys crates, are set
+# below instead. CC/CXX are used by some crates to compile temporary binaries to
+# run on the host system during the build, so using a cross compiler there is
+# not something we currently want. However, AR, for example, is used by `ring`
+# to prepare a library for the target system, so we don't reset it here. This
```
[the remark this answers is not in the document]
Comment (aqrln) [reply cross/linux-musl-arm.Dockerfile:82]: example of how it failed with `ENV AR=`: https://buildkite.com/prisma/release-prisma-engines/builds/2807#_

**Review by jkomyno — APPROVED** [review]
I've tested this locally and in [ecosystem tests](https://github.com/prisma/ecosystem-tests/pull/3418), and it works on Alpine 3:16 and 3:17.
While there may be improvements to how we currently compile these new builds, I deem it safe to ship this in `prisma@4.10.0` and then, potentially, revisit later once Julius is back

File: cross/linux-musl-arm.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    elif [ "$OPENSSL_VARIANT" = "1.1.x" ]; then \
+        OPENSSL_VERSION="$OPENSSL_1_1_VERSION"; \
+    else \
+        >&2 echo "wrong openssl variant: $OPENSSL_VARIANT"; \
+        exit 1; \
+    fi && \
+    curl -fLO https://www.openssl.org/source/openssl-$OPENSSL_VERSION.tar.gz && \
+    tar xzf openssl-$OPENSSL_VERSION.tar.gz && \
+    cd openssl-$OPENSSL_VERSION && \
+    ./Configure -static zlib linux-aarch64 --prefix=/opt/cross --openssldir=/opt/cross && \
```
Comment (aqrln) [inline cross/linux-musl-arm.Dockerfile:71]: I’d really prefer to build a shared library here, but I had some issues with it and decided to use a static library for now instead to make it work first. It is also consistent with our existing glibc binaries for ARM which also statically link OpenSSL: https://github.com/prisma/engine-images/blob/main/cross/linux-arm-ssl-3.0.x.Dockerfile

In the long run both should optimally use shared libraries, but I’d rather not spend too much time on it right now.

**Pull request merged**